OMPython/__init__.py

```diff
@@ -447,3 +462,727 @@ def getClassNames(self, className=None, recursive=False, qualified=False, sort=F
+			else:
+				print 'loadFile Error: ', loadfileError
+			return
+			
+		if lmodel is None:
+			pass
+		else:
+			loadmodelError = ''
+			loadModelResult = self.requestApi("loadModel", lmodel)		
+			loadmodelError = self.requestApi('getErrorString')
+			if loadmodelError:
+				print loadmodelError
+				return
+			
+		buildModelError = ''
+		self.getconn.sendExpression("setCommandLineOptions(\"+d=initialization\")")
+		buildModelResult = self.requestApi("buildModel", mName)
+		buildModelError = self.requestApi("getErrorString")
+		if buildModelError:
+			if 'Message: Fault' in buildModelError:
+				print ''
+			else:
+				print 'buildModel Error: ', buildModelError
+				return
+		
+		self.xmlFile = buildModelResult[1]
+		self.tree = ET.parse(self.xmlFile)
+		self.root = self.tree.getroot()
+		self.createQuantitiesList()
+		self.getQuantitiesNames()
+		self.getContinuousNames()
+		self.getParameterNames()
+		self.inputs = self.getInputNames()
+		self.setInputSize()
+		self.getOutputNames()
+		self.getContinuousValues()
+		self.getParameterValues()
+		self.getInputValues()
+		self.getOutputValues()
+		self.getSimulationValue()
+		return
+
+	def requestApi(self, apiName, entity=None, properties=None ):
+		if (entity is not None and properties is not None):
+			exp = '{}({}, {})'.format(apiName, entity, properties)
+		elif entity is not None and properties is None:
+			if apiName == "loadFile" or apiName == "importFMU":
+				exp = '{}("{}")'.format(apiName, entity)
+			else:
+				exp = '{}({})'.format(apiName, entity)
+		else:
+
+			exp = '{}()'.format(apiName)
+		try:
+			res = self.getconn.sendExpression(exp)
+		except pyparsing.ParseException as e:
+			res = 'Message: Fault'
+		return res
+		
+		
+	def requestApi__(self, apiName, entity ):
+			
+		exp = '{}({})'.format(apiName, entity)
+		
+		res = self.getconn.sendExpression(exp)
+			
+		return res
+
+	def createQuantitiesList(self):
+		rootCQ = self.root
+		if self.quantitiesList == []:
+			for sv in rootCQ.iter('ScalarVariable'):
+				name = sv.get('name')
+				changable = sv.get('isValueChangeable')
+				description = sv.get('description')
+				variability = sv.get('variability')
+				causality = sv.get('causality')
+				ch = sv.getchildren()
+				start = None
+				for att in ch:
+					start = att.get('start')
+				self.quantitiesList.append(Quantity(name, start, changable, variability, description, causality))
+		return self.quantitiesList
+		
+	def getQuantitiesNames(self):
+		if self.qNamesList == []:
+			for q in self.quantitiesList:
+				self.qNamesList.append(q.name)
+		else:
+			pass
+		return self.qNamesList
+		
+	def checkAvailability(self, names, chkList, inputFlag = None):
+		isExist = True
+		if inputFlag is not None:
+			if names in chkList:
+				pass
+			else:
+				isExist = False
+			if(isExist == False):
+				print 'Error!!! ', names , ' does not exist.'
+				return False
+		
+		else:
+			nonExistingList = []
+			for n in names:
+				if n in chkList:
+					pass
```

Comment:
again an empty if/else clause. from now on i'll just assume this is a bad coding pattern. (would be good to learn to avoid doing that, tbh)